validator/project_validator.go

```diff
@@ -618,6 +618,35 @@ func checkProjectFields(project *model.Project) ValidationErrors {
 	return errs
 }
 
+func validateBuildVariantTaskNames(task string, variant string, allTaskNames map[string]bool, taskGroupTaskSet map[string]string) []ValidationError {
+	var errs []ValidationError
+	if _, ok := allTaskNames[task]; !ok {
+		if task == "" {
+			errs = append(errs, ValidationError{
+				Message: fmt.Sprintf("tasks for buildvariant '%s' must each have a name field",
+					variant),
+				Level: Error,
+			})
+
+		} else {
+			errs = append(errs, ValidationError{
+				Message: fmt.Sprintf("buildvariant '%s' references a non-existent task '%s'",
+					variant, task),
+				Level: Error,
+			})
+		}
+	}
+	if _, ok := taskGroupTaskSet[task]; ok {
+		errs = append(errs,
+			ValidationError{
+				Message: fmt.Sprintf("task '%s' in build variant '%s' is already referenced in task group '%s'",
```

Comment:
Given the rational for this error in the original ticket EVG-6846, I don't think we'd want to give this warning for inline task group tasks passed in, since they _are _correctly being used in a task group.

Reply:
Good catch! Fixing.